From 5f1327a88e9ae8fd1c1c92f7498fb6a0a2624a27 Mon Sep 17 00:00:00 2001
From: Axel Spoerl <axel.spoerl@qt.io>
Date: Tue, 17 Jun 2025 14:07:32 +0200
Subject: [PATCH 28/31] QMainWindowTabBar::dockAt() - avoid nullptr access

If the main window layout was already deleted, the method crashed with
a nullptr access to the return value of qt_mainwindow_layout().

Handle a nullptr return value properly.

Task-number: QTBUG-147755
Change-Id: I991e831f58e97267a518a98f032f8aa7ea0384bd
Reviewed-by: Samuel Gaist <samuel.gaist@idiap.ch>
(cherry picked from commit cc1ddebf63a2b6c051529d1d3127534aa4c1d55a)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
(cherry picked from commit a63b47a7a6bef5ffcce5b5bdcfceda34faf0ee4b)
---
 src/widgets/widgets/qmainwindowlayout.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/widgets/widgets/qmainwindowlayout.cpp b/src/widgets/widgets/qmainwindowlayout.cpp
index f46095abc86..dab473a900d 100644
--- a/src/widgets/widgets/qmainwindowlayout.cpp
+++ b/src/widgets/widgets/qmainwindowlayout.cpp
@@ -1953,7 +1953,7 @@ QDockWidget *QMainWindowTabBar::dockAt(int index) const
 {
     QMainWindowTabBar *that = const_cast<QMainWindowTabBar *>(this);
     QMainWindowLayout* mlayout = qt_mainwindow_layout(mainWindow);
-    QDockAreaLayoutInfo *info = mlayout->dockInfo(that);
+    QDockAreaLayoutInfo *info = mlayout ? mlayout->dockInfo(that) : nullptr;
     if (!info)
         return nullptr;
 
-- 
2.50.1

